### DISCUSSION TITLE
Percent of image tolerance per object

### DISCUSSION THREAD

**Discussion by Rotox** [VorlonCD/bi-aidetection#229] [Ideas]
Not sure if there's a way to do this currently or maybe an idea for a feature. Would like to be able to set tolerances on percent of image but per object as a way to filter out false positives. For example one of my cameras would never see a person as larger than 3% of the image whereas a car or truck could be 10% or higher. I know you can set the percent of image tolerances per camera but that would then filter out actual cars and trucks. Ive had a couple of bugs on the lens and the side of the house trigger as a person even though they took up over 10% of the image. So being able to set percent of image tolerances per object would be a great way to eliminate those

**Comment by VorlonCD** [comment]
That was put in my last update yesterday:
https://github.com/VorlonCD/bi-aidetection/tree/master/src/UI/Installer

**Reply by Rotox** [reply]
Ah great timing on my question then! Just downloaded it and setting it up now. That's great to know that .5 can be used as a value especially for the smaller ones in our family. Will test it out on a few of our cameras. Thanks!

**Reply by Rotox** [reply]
I left the minimum percentage at 0% and adjusted the maximum but its marking anything with values of 0.2-0.5 as "TooSmallPercent". Seems to be working good with anything 1% and higher. I figured leaving it at 0% would catch everything on the low end as I'm really only looking to eliminate the bigger objects. Or do you have to specify a value like 0.5? If so can a lower value like 0.1 or 0.2 be used?

**Reply by VorlonCD** [reply]
Yes, you can use any decimal amount above 0.  0 should allow anything through... You are saying that is not true?

If you look on History tab > Prediction Details > "PercentOfImage" column you will see the current size of each object.   Sometimes that number can be deceptively small so you have to keep an eye on it.

If you are seeing "TooSmallPercent" that means you adjusted the Min/max size in CAMERAS Tab > Predictions Tolerances.   However, since you want to control by object you should be changing the number in CAMERAS Tab > Relevant Objects > select object > "Limit Size Percentage" > Min / Max.

If you dont think it is behaving correctly, I may need screenshots of Prediction Details screen and log entries from near the time the object in question was processed.

**Comment by Rotox** [comment]
I checked and I definitely have it set on the object level as you mentioned. I'm attaching a screenshot of the prediction details for one of the TooSmallPercent failures as well as a screenshot of the relevant logs near that time the object was processed. Hope that helps and let me know if you need any additional information
<img width="472" alt="Screen Shot 2021-04-10 at 4 43 57 PM" src="https://user-images.githubusercontent.com/6833008/114284267-a69cfc80-9a1c-11eb-875c-bab7922bde80.png">
<img width="1674" alt="Screen Shot 2021-04-10 at 4 46 40 PM" src="https://user-images.githubusercontent.com/6833008/114284270-af8dce00-9a1c-11eb-9261-0467f2e330bb.png">

**Reply by VorlonCD** [reply]
screenshot of Camera > Prediction Tolerances

**Comment by Rotox** [comment]
<img width="294" alt="Screen Shot 2021-04-10 at 5 05 43 PM" src="https://user-images.githubusercontent.com/6833008/114284617-1613eb80-9a1f-11eb-97f2-986074135a04.png">

**Reply by VorlonCD** [reply]
Think of **Prediction Tolerances** for the camera as the gatekeeper - Nothing will get past these settings.

So you have to set MIN to `.4` or less here, **THEN the Relevant Objects screen can further refine what is a valid object.**

![image](https://user-images.githubusercontent.com/7512113/114285091-68a2d700-9a22-11eb-87f2-40614ff728e9.png)

![image](https://user-images.githubusercontent.com/7512113/114285101-82dcb500-9a22-11eb-8a8e-6297477e121f.png)

**Comment by Rotox** [comment]
Ah ok that makes total sense. I readjusted the camera tolerance to 0% and set the person object to be 0.1% so will see how that works out. Appreciate the explanation!
